src/pages/index.js

```diff
@@ -104,50 +182,118 @@ function createCard(data) {
   return cardEl;
 }
 
-// Render initial cards
-initialCards.forEach(card => {
-  cardsList.appendChild(createCard(card));
-});
+function setUserInfo(user) {
+  profileName.textContent = user.name;
+  profileTitle.textContent = user.about;
+  profileAvatar.src = user.avatar;
+}
+
+// Edit profile form
+function handleEditProfileSubmit(data, submitButton) {
+  setButtonText(submitButton, true, 'Saving...', 'Save');
+  api.editUserInfo(data)
+    .then(updatedUser => {
+      setUserInfo(updatedUser);
+      closeModal(editProfileModal);
+    })
+    .catch(err => console.error('Edit profile error:', err))
+    .finally(() => {
+      setButtonText(submitButton, false, 'Saving...', 'Save');
+    });
+}
+
+// Update avatar
+function handleEditAvatarSubmit(avatarUrl, submitButton) {
+  setButtonText(submitButton, true, 'Saving...', 'Save');
+  api.updateUserAvatar(avatarUrl)
+    .then(updatedUser => {
+      setUserInfo(updatedUser);
+      closeModal(editAvatarModal);
+    })
+    .catch(err => console.error('Edit avatar error:', err))
+    .finally(() => {
+      setButtonText(submitButton, false, 'Saving...', 'Save');
+    });
+}
 
-// Form handlers
+// Initial load
+api.getAppInfo()
+  .then(([cards, user]) => {
+    setUserInfo(user);
+    cards.forEach(card => {
+      cardsList.append(createCard(card));
+    });
+  })
+  .catch(err => console.error('App init error:', err));
+
+// 📝 Form Listeners
+
+// Edit profile
 document.getElementById('edit-profile-form').addEventListener('submit', e => {
   e.preventDefault();
-  profileName.textContent = profileNameInput.value;
-  profileTitle.textContent = profileDescriptionInput.value;
-  closeModal(editProfileModal);
+  const submitButton = e.submitter;
+  const updated = {
+    name: profileNameInput.value,
+    about: profileDescriptionInput.value
+  };
+  handleEditProfileSubmit(updated, submitButton);
 });
 
+// Add new card
 document.getElementById('new-post-form').addEventListener('submit', e => {
   e.preventDefault();
+  const submitButton = e.submitter;
   const newCard = {
     name: cardCaptionInput.value,
     link: cardImageInput.value
   };
-  cardsList.prepend(createCard(newCard));
+
+  setButtonText(submitButton, true, 'Saving...', 'Create');
+  api.addCard(newCard)
+    .then(card => {
+      cardsList.prepend(createCard(card));
+      closeModal(newPostModal);
+      e.target.reset();
```

Comment:
Please, **disable** the save button in `.then` after submission, according to the previous task